#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = 'control_sys'

gen = ParameterGenerator()
gen.add('roll_mixer', double_t, 0, 'roll rate controller output is multiplied by this', 1, 0, 10)
gen.add('pitch_mixer', double_t, 0, 'pitch rate controller output is multiplied by this', 1, 0, 10)
gen.add('yaw_mixer', double_t, 0, 'yaw rate controller output is multiplied by this', 1, 0, 10)
gen.add('z_mixer', double_t, 0, 'z (altitude) controller output is multiplied by this', 1, 0, 10)

gen.add('roll_p', double_t, 0, 'roll rate controller output is multiplied by this', 1, 0, 10)
gen.add('pitch_p', double_t, 0, 'pitch rate controller output is multiplied by this', 1, 0, 10)
gen.add('yaw_p', double_t, 0, 'yaw rate controller output is multiplied by this', 5, 0, 10)

gen.add('motor_on', bool_t, 0, 'If unchecked, send 0 commands to motors', False)
gen.add('motor_init', bool_t, 0, 'sets the motor PWM to the minimum value required for starting them. Only use after turning on (motor_on)', True)

gen.add('throttle_bias', int_t, 0, 'Default value is required for hover', 170, 166, 255)

exit(gen.generate(PACKAGE, 'mixer', 'mixer_param'))
